baseapp/oe/optimistic_execution.go

```diff
@@ -0,0 +1,174 @@
+package oe
+
+import (
+	"bytes"
+	"math/rand"
+	"sync"
+	"time"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"cosmossdk.io/log"
+)
+
+type OptimisticExecution struct {
+	mtx         sync.RWMutex
+	stopCh      chan struct{}
+	shouldAbort bool
+	running     bool
+	initialized bool
+
+	// we could use generics here in the future to allow other types of req/resp
+	fn            func(*abci.RequestFinalizeBlock) (*abci.ResponseFinalizeBlock, error)
```

Comment:
```suggestion
	finalizeBlockExecFn            func(*abci.RequestFinalizeBlock) (*abci.ResponseFinalizeBlock, error)
```
^^ or something more descriptive.